### PR TITLE
Implement cleanup methods to cleanup underlying cluster resources in federated service tests.

### PR DESCRIPTION
**Pull request by madhusudancs**
cc @kubernetes/sig-cluster-federation @colhom

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32810)

### CODE REVIEW

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

File: test/e2e/federated-service.go [hunk 1]
```diff
@@ -80,10 +80,14 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 				nsName := f.FederationNamespace.Name
 				service := createServiceOrFail(f.FederationClientset_1_4, nsName, FederatedServiceName)
 				By(fmt.Sprintf("Creation of service %q in namespace %q succeeded.  Deleting service.", service.Name, nsName))
+
 				// Cleanup
 				err := f.FederationClientset_1_4.Services(nsName).Delete(service.Name, &api.DeleteOptions{})
 				framework.ExpectNoError(err, "Error deleting service %q in namespace %q", service.Name, service.Namespace)
 				By(fmt.Sprintf("Deletion of service %q in namespace %q succeeded.", service.Name, nsName))
+
+				By(fmt.Sprintf("Deleting service shards and their provider resources in underlying clusters for service %q in namespace %q", service.Name, nsName))
```
Comment (nikhiljindal) [inline test/e2e/federated-service.go]: Move this to AfterEach so that it always runs?

**Review by nikhiljindal — CHANGES_REQUESTED** [review]

**Comment by nikhiljindal** [comment]
Also do you want to do the same for ingress test as well?

**Comment by nikhiljindal** [comment]
That new red cross for changes requested is really big!

File: test/e2e/federated-service.go (hunk 1 above)
Comment (madhusudancs) [inline test/e2e/federated-service.go]: Done.

**Review by madhusudancs — COMMENTED** [review]
> Also do you want to do the same for ingress test as well?

Yeah, but in a follow up PR.

**Comment by madhusudancs** [comment]
@nikhiljindal Addressed your comments. PTAL.

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Comment by nikhiljindal** [comment]
@madhusudancs I assume you want this cherrypicked as well?
Otherwise the new release-1.4 tests that you have setup will keep failing without this.

**Comment by madhusudancs** [comment]
@nikhiljindal I would like this to be cherry-picked. But I don't mind if it is not cherry-picked right now, i.e. for 1.4.0. It could be cherry-picked later for one of the patch releases.

**Pull request merged**